pkg/github/token.go

```diff
@@ -131,5 +184,28 @@ func (g GitHubTokenClient) GetNewGitHubClient(token string) (GitHubClient, error
 		Client:    client,
 	}
 
-	return githubClient, nil
+	return &githubClient, nil
+}
+
+func rateLimitCallBackfunc(cbContext *github_ratelimit.CallbackContext) {
+	// Retrieve the request's context and get the client name from it
+	// Use the client name to lookup the client pointer
+	req := *cbContext.Request
+	reqCtx := req.Context()
+	ghClientName := reqCtx.Value(GHClientKey).(string)
+	ghClient := Clients[ghClientName]
+
+	// Start a goroutine that marks the given client as rate limited and sleeps for 'TotalSleepTime'
+	go func(client *GitHubClient) {
+		client.SecondaryRateLimit.mu.Lock()
+		client.SecondaryRateLimit.isLimitReached = true
+		client.SecondaryRateLimit.mu.Unlock()
+
+		// Sleep until the rate limit is over
+		time.Sleep(*cbContext.TotalSleepTime)
```

Comment:
I think you want to use cbContext.SleepUntil here, as TotalSleepTime accumulates the sleep durations in case of multiple hits of secondary rate limit.
So probably looking for:
`time.Sleep(time.Until(*cbContext.SleepUntil))` 

I'll add documentation for the different fields in go-github-ratelimit to avoid this kind of confusion.

Reply:
Hmm, interesting. In my tests, I'm finding that `time.Until(*cbContext.SleepUntil)` is a negative number, and the callback function is already sleeping before I call my `time.Sleep` 🤔